src/Rings/mpoly-graded.jl

```diff
@@ -1612,6 +1612,349 @@ function Base.show(io::IO, h::HilbertData)
   print(io, "Hilbert Series for $(h.I), data: $(h.data), weights: $(h.weights)")  ###new
 end
 
+########################################################################
+# Compute the Hilbert series from the monomial leading ideal using 
+# different bisection strategies along the lines of 
+# [Kreuzer, Robbiano: Computational Commutative Algebra 2, Springer]
+# Section 5.3.
+########################################################################
+exp_vec(f::MPolyElem) = first(exponent_vectors(f))
+
+_divides(a::Vector{Int}, b::Vector{Int}) = all(x->(x>=0), a-b)
+
+function _gcd(a::Vector{Vector{Int}}) 
+  length(a) == 1 && return a[1]
+  n = length(a[1]) # assumed to be the length of all entries of a
+  return [minimum([a[i][j] for i in 1:length(a)]) for j in 1:n]
+end
+
+function _are_pairwise_coprime(a::Vector{Vector{Int}})
+  length(a) <= 1 && return true
+  n = length(a)
+  for i in 1:n-1
+    for j in i+1:n
+      all(x->(x==0), _gcd([a[i], a[j]])) || return false
+    end
+  end
+  return true
+end
+
+### Assume that a is minimal. We divide by `pivot` and return 
+# a minimal set of generators for the resulting monomial ideal.
+function _divide_by(a::Vector{Vector{Int}}, pivot::Vector{Int})
+  shifted = [e-pivot for e in a]
+  # The good ones will contribute to a minimal generating 
+  # set of the lhs ideal.
+  good = [e for e in shifted if all(x->(x>=0), e)]
+
+  # The bad monomials come from those which hop over the boundaries of 
+  # the monomial diagram by the shift. Their span has a new 
+  # generator which is collected in `bad` a priori. It is checked 
+  # whether they become superfluous and if not, they are added to 
+  # the good ones.
+  bad = [e for e in shifted if !all(x->(x>=0), e)]
+
+  divides(a::Vector{Int}, b::Vector{Int}) = all(k->(k>=0), b - a)
+
+  for e in bad 
+    m = [k < 0 ? 0 : k for k in e]
+    if all(x->(!divides(x, m)), good)
+      good = [x for x in good if !divides(m, x)]
```

Comment:
This is a point where keeping those Int64 bit masks as a pre-divisibility test, as described by @hannes14, would probably have a major effect.